clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -630,6 +630,12 @@ public CreateBucketResponseType createBucket(final CreateBucketType request) thr
         throw new InvalidBucketNameException(request.getBucket());
       }
 
+      if (Principals.isFakeIdentify(canonicalId)) {
```

Comment:
(Adding from @sjones4 review)

> I would agree that we should not put authorization checks (i.e. Principals.isFakeIdentify) in the ObjectStorageGateway but rather in OsgAuthorizationHandler.

The only reason this check was in ObjectStorageGateway was because of the strange logic already there, where it creates a bucket with the requesting user as owner (nobody, in this case), and then sees if that operation is allowed via OsgAuthorizationHandler's operationAllowed(), which will always return true.

We could reorg it better to remove the check from ObjectStorageGateway, to explicitly deny bucket creation by nobody, but allowing object creation (if the bucket ACP allows it).